### ISSUE TITLE
Cannot autostart

### ISSUE THREAD

**Issue by Yajo** [flathub/im.riot.Riot#231]
There's an option in Element to autostart on session login:

![imagen](https://user-images.githubusercontent.com/973709/140638316-c4d78220-ed9e-414f-8a7f-4391f2f5419d.png)

It doesn't work in the flatpak version. You can enable it, but then log out and log in your desktop session, and you'll see that you have to open Element again, and if you go to that option, you'll see it disabled again.

**Comment by SISheogorath** [comment]
Yes, flatpaks can't automate their own startup. Since the flatpak has no access to the autoatart directories of your home directory it can not put the desktop file there that would trigger the autostart.

Further even if it would be able to, it would write a wrong file in there, since the command has to run outside of the sandbox and is different from that the flatpak sees inside.

To sum it up, this is wanted behaviour because it would:

1. Break the sandbox of flatpak
2. Result in a broken file that wouldn't woek anyway

**Comment by SISheogorath** [comment]
Note: if you want to manage autostart programs safely on GNOME, you can use GNOME Tweaks for that: https://help.gnome.org/users/gnome-help/stable/shell-apps-auto-start.html.en

KDE has a similar tool built i to it's settings, as far as I know.

If you use yet another desktop environment, please refer to their documentation.

**Comment by Yajo** [comment]
But I use the dejadup flatpak and it starts automatically on login to do its automated backups...

**Comment by SISheogorath** [comment]
Fair point, I did some investigation into how deja-dub does it, and it seems there is a portal for managing autostart. Sadly, this will mean that Electron or a separate library used with it, will have to implement it upstream, which at least by my research I couldn't even find a feature request for.

https://github.com/flatpak/xdg-desktop-portal/blob/135133e87aff92066ae1e49a4e6d2a4b8b92a9d7/data/org.freedesktop.impl.portal.Background.xml#L97-L103

So I guess there is not much we can do about that currently, the flatpak side, the portal, is actually already there. So it up to the waiting game regarding adoption.

**Comment by Yajo** [comment]
Then it must be fixed upstream, right? However, upstream doesn't officially support flatpak, so how do you think would be the best way to proceed?

**Comment by vchernin** [comment]
The correct solution is for upstream to use the portal. The libportal library can help. Not sure if that's best in Electron or Element though. Note the background portal doesn't currently work fully on the host, so the portal would be flatpak only for now.

**Comment by bijavix** [comment]
Still a very annoying issue, since flatpak apps are capable of creating the startup entry. 

In the meantime _(if this ever gets fixed)_, you can manually create the startup entry to launch element and minimise it automatically:

(Works with GNOME X11)

Install `xdotool`, then create `~/.config/autostart/im.riot.Riot.desktop`:

```
[Desktop Entry]
Type=Application
Exec=sh -c 'flatpak run im.riot.Riot & sleep 5 && xdotool search --sync --onlyvisible --classname ".*Element.*" windowminimize'
NoDisplay=false
X-GNOME-Autostart-enabled=true
Name=im.riot.Riot
Comment=matrix client
```
